public/pages/Main/Main.tsx

```diff
@@ -32,62 +44,155 @@ enum Pathname {
   Channels = '/channels',
 }
 
-interface MainProps extends RouteComponentProps {}
+interface MainProps extends RouteComponentProps {
+  setActionMenu: (menuMount: MountPoint | undefined) => void;
+  multiDataSourceEnabled: boolean;
+  dataSourceManagement: DataSourceManagementPluginSetup;
+}
 
-export interface MainState {
+export interface MainState extends Pick<DataSourceMenuProperties, "dataSourceId" | "dataSourceLabel"> {
   availableChannels: Partial<typeof CHANNEL_TYPE>;
   availableConfigTypes: string[]; // available backend config types
   tooltipSupport: boolean; // if true, IAM role for SNS is optional and helper text should be available
+  dataSourceReadOnly: boolean;
+  dataSourceLabel: string;
 }
 
 export const MainContext = createContext<MainState | null>(null);
 
 export default class Main extends Component<MainProps, MainState> {
   static contextType = ServicesContext;
-
   constructor(props: MainProps) {
     super(props);
-    this.state = {
-      availableChannels: CHANNEL_TYPE,
-      availableConfigTypes: [],
-      tooltipSupport: false,
-    };
+    let dataSourceId = "";
+    let dataSourceLabel = "";
+    if (props.multiDataSourceEnabled) {
+      const {
+        dataSourceId: parsedDataSourceId,
+        dataSourceLabel: parsedDataSourceLabel
+      } = queryString.parse(this.props.location.search) as {
+        dataSourceId: string;
+        dataSourceLabel: string;
+      };
+      dataSourceId = parsedDataSourceId || "";
+      dataSourceLabel = parsedDataSourceLabel || "";
+      this.state = {
+        dataSourceId: dataSourceId,
+        dataSourceLabel: dataSourceLabel,
+        dataSourceReadOnly: false,
+        availableChannels: CHANNEL_TYPE,
+        availableConfigTypes: [],
+        tooltipSupport: false,
+      };
+    } else {
+      this.state = {
+        availableChannels: CHANNEL_TYPE,
+        availableConfigTypes: [],
+        tooltipSupport: false,
+      };
+    }
   }
 
   async componentDidMount() {
-    const serverFeatures = await this.context.notificationService.getServerFeatures();
-    if (serverFeatures != null) {
-      this.setState({
-        availableChannels: serverFeatures.availableChannels,
-        availableConfigTypes: serverFeatures.availableConfigTypes,
-        tooltipSupport: serverFeatures.tooltipSupport,
-      });
-    } else {
-      // Feature API call failed, allow all configs to avoid UI breaking.
-      // User requests will still be validated by backend.
+
+    const services = this.getServices(this.props.http) // Assuming this.context holds the value provided by ServicesContext
+    const serverFeatures = await services.notificationService.getServerFeatures();
+
+    if (this.props.multiDataSourceEnabled) {
```

Comment:
Nitpick:
Looks like the only difference between the `if/else` cases here is whether `dataSourceId` gets added to the state. Rather than have 2 large `if/else` blocks, could we just conditionally add the `dataSourceId`?

Reply:
addressed